src/common/js_logger.hpp

```diff
@@ -0,0 +1,131 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+#include <iostream>
+#include "sync/sync_manager.hpp"
+
+#if REALM_ANDROID
+    #include <android/log.h>
+#endif
+
+namespace realm {
+namespace common {
+
+using logger_fn = std::function<void(std::string, std::string)>;
+using log_level = realm::util::Logger::Level;
+using log_entry = std::pair<std::string, std::string>;
```

Comment:
I'm not particularly well versed in C++ code styles, but I would expect these types to be PascalCased.